File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    # goal 배열을 순차적으로 탐색하면서
    for i in goal:
        # 만약 i가 cards1의 0번 인덱스에 있으면 cards1의 0번 인덱스를 빼주고
        if cards1 and i == cards1[0]:
            cards1.pop(0)
        # 만약 i가 cards2에 있으면 cards2의 0번 인덱스를 빼준다.
        elif cards2 and i == cards2[0]:
            cards2.pop(0)
        # 그러다 위의 두 경우에 해당하지 않게 되면 'No를 반환한다'
        else:
            return 'No'
    
    return 'Yes'

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int idx1 = 0; // cards1의 인덱스 정보 저장
        int idx2 = 0; // cards2의 인덱스 정보 저장
        
        // goal 배열만큼 탐색하면서
        for (int i = 0; i < goal.length; i++) {
            // idx1이 cards1의 길이보다 작으면 goal의 i번 인덱스와 cards1의 idx1번 인덱스를 비교해서
            if (idx1 < cards1.length && goal[i].equals(cards1[idx1])) {
                idx1++; // 같으면 idx1을 증가
            // idx2가 cards2의 길이보다 작으면 goal의 i번 인덱스와 cards2의 idx2번 인덱스를 비교해서
            } else if (idx2 < cards2.length && goal[i].equals(cards2[idx2])) {
                idx2++; // 같으면 idx2를 증가
            } else { // 위의 조건에 해당하지 않으면 'No' 반환하고 끝
                return "No";
            }
        }
        // 전체 조건에 문제가 없었다면 'Yes' 반환
        return "Yes";
    }
}